src/Oci8/Query/Grammars/OracleGrammar.php

```diff
@@ -494,4 +494,65 @@ protected function dateBasedWhere($type, Builder $query, $where)
 
         return "extract ($type from {$this->wrap($where['column'])}) {$where['operator']} $value";
     }
+
+    /**
+     * Compile a "where not in raw" clause.
+     *
+     * For safety, whereIntegerInRaw ensures this method is only used with integer values.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @param  array  $where
+     * @return string
+     */
+    protected function whereNotInRaw(Builder $query, $where)
+    {
+        if (! empty($where['values'])) {
+            if (is_array($where['values']) && count($where['values']) > 1000) {
+                return $this->resolveClause($where['column'], $where['values'], 'not in');
+            } else {
+                return $this->wrap($where['column']).' not in ('.implode(', ', $where['values']).')';
+            }
+        }
+
+        return '1 = 1';
+    }
+
+    /**
+     * Compile a "where in raw" clause.
+     *
+     * For safety, whereIntegerInRaw ensures this method is only used with integer values.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @param  array  $where
+     * @return string
+     */
+    protected function whereInRaw(Builder $query, $where)
+    {
+        if (! empty($where['values'])) {
+            if (is_array($where['values']) && count($where['values']) > 1000) {
+                return $this->resolveClause($where['column'], $where['values'], 'in');
+            } else {
+                return $this->wrap($where['column']).' in ('.implode(', ', $where['values']).')';
+            }
+        }
+
+        return '0 = 1';
+    }
+
+    private function resolveClause($column, $values, $type)
+    {
+        $chunks = array_chunk($values, 1000);
+        $whereClause = '';
+        $i=0;
+        $type = $this->wrap($column) . ' '.$type.' ';
+        foreach ($chunks as $ch) {
```

Comment:
I think this should be inside a where closure like the implementation below:

https://github.com/yajra/laravel-oci8/blob/1195c472717833474657382b3e8f50b7e9230854/src/Oci8/Query/OracleBuilder.php#L80-L85

Reply:
This part is made based on previous code and I don't see why change it. Feel free to modify whatever you considerate to make it work.

Reply:
This part is made based on previous code, your method doesn't work on Grammar. Please feel free to modify whatever you considerate to make it work.

Reply:
I see, I think the issue will occur when you have additional where clause on your relation?

```php
Model:where('1', '=', '1')
->with(['relation' => function($q) {
   $q->where('active', 1);
})
->get();
```

It will make:

```sql
select * from relation 
where id in ("results of last query") 
  or id in("next 1000 set")
  and active = 1;
```

Reply:
Ok I will fix that with parenthesis